pyfpga/diamond.py

```diff
@@ -0,0 +1,30 @@
+#
+# Copyright (C) 2024 PyFPGA Project
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+
+"""
+Implements support for Diamond.
+"""
+
+import os
+from pyfpga.project import Project
+
+
+class Diamond(Project):
+    """Class to support Diamond projects."""
+
+    def _configure(self):
+        tool = 'diamond'
+        executable = 'pnmainc' if os.name == 'nt' else 'diamondc'
+        self.conf['tool'] = tool
+        self.conf['make_cmd'] = f'{executable} {tool}.tcl'
```

Comment:
According to the Diamond UG 3.9 & 3.12, TCL should be prefixed with -t:

> To launch Diamond and run a Tcl script:
> In the Tcl console or other command line shell, enter the following command:
> On Windows:
> pnmain.exe -t<tcl_path_file>
> On Linux:
> diamond -t<tcl_path_file>

Source: Lattice Diamond 3.12 User Guide p.137

diff --git a/pyfpga/diamond.py b/pyfpga/diamond.py
index a5b3e62..49a5d8d 100644
--- a/pyfpga/diamond.py
+++ b/pyfpga/diamond.py
@@ -19,7 +19,7 @@ class Diamond(Project):
         tool = 'diamond'
         executable = 'pnmainc' if os.name == 'nt' else 'diamondc'
         self.conf['tool'] = tool
-        self.conf['make_cmd'] = f'{executable} {tool}.tcl'
+        self.conf['make_cmd'] = f'{executable} -t{tool}.tcl'
         self.conf['make_ext'] = 'tcl'
         self.conf['prog_bit'] = 'bit'
         self.conf['prog_cmd'] = f'sh {tool}-prog.sh'

Reply:
Hmm this is interesting. When I add the `-t` flag in Linux, it doesn't run the script and waits for input on stdin (which it never gets). Did you test whether -t works on Windows?

Reply:
I understood that it was working before the **-t** addition. Maybe not needed.

Reply:
After another check (and in contrary to the documentation), the -t seems to be necessary for pnmain (GUI) but not pnmainc (CLI) on Windows. Remvoe the -t from the PR and we will open a support case with Lattice.

Reply:
Removed (never added).